### PR TITLE
#164798299 Users should be able to receive notifications 

### PR DESCRIPTION
**Pull request by anyatibrian**
### What does this PR do?
Enable authenticated user to receive notifications when users they follow publish new articles.
Enable users receive notifications when articles they favorited have new interactions (like new comments).
### Description of Task to be completed?

- Add a URL for viewing all notifications
- Add a URL for viewing unread notifications
- Add a notification into the database
- send email notifications to users
- Add a URL for viewing single notification

### How should this be manually tested?
when you hit this endpoint `localhost:8000/api/v1/users/notifications/` 
this is what is expected 
```
{
    "notifications": {
        "notificationsCount": 20,
        "notifications": [
            {
                "id": 10,
                "receiver": 2,
                "message": "anyatibrian comment on the goneway chiled at kamukya\n with \n this is some awsome article",
                "link": null,
                "is_read": false,
                "created_at": "2019-04-17T15:39:54.562841Z"
            },
            {
                "id": 11,
                "receiver": 2,
                "message": "anyatibrian comment on the goneway chiled at kamukya\n with \n Lorem ipsum dolor sit amet, consectetur",
                "link": null,
                "is_read": false,
                "created_at": "2019-04-17T15:55:24.267592Z"
            }
}
}
```
when you hit this endpoint `localhost:8000/api/v1/users/notifications/unread/`

```
{
    "notifications": {
        "notificationCounts": 20,
        "notifications": [
            {
                "id": 10,
                "receiver": 2,
                "message": "anyatibrian comment on the goneway chiled at kamukya\n with \n this is some awsome article",
                "link": null,
                "is_read": false,
                "created_at": "2019-04-17T15:39:54.562841Z"
            },
            {
                "id": 11,
                "receiver": 2,
                "message": "anyatibrian comment on the goneway chiled at kamukya\n with \n Lorem ipsum dolor sit amet, consectetur",
                "link": null,
                "is_read": false,
                "created_at": "2019-04-17T15:55:24.267592Z"
            },
            {
                "id": 12,
                "receiver": 2,
                "message": "anyatibrian comment on the goneway chiled at kamukya\n with \n Lorem ipsum dolor sit amet",
                "link": null,
                "is_read": false,
                "created_at": "2019-04-17T15:56:18.512102Z"
            }
}
}

```
when you hit this endpoint `localhost:8000/api/v1/users/notifications/10/unread/`
```
{
    "notifications": [
        {
            "id": 10,
            "receiver": 2,
            "message": "anyatibrian comment on the goneway chiled at kamukya\n with \n this is some awsome article",
            "link": null,
            "is_read": true,
            "created_at": "2019-04-17T15:39:54.562841Z"
        }
    ]
}
```
### Relevant Pivotal Tracker Stories 
[#164798299](https://www.pivotaltracker.com/story/show/164798299)
### Screen Shorts
Screenshot 2019-04-18 at 00 10 22
Screenshot 2019-04-18 at 00 08 06

### CODE REVIEW

File: authors/apps/articles/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
```
Comment (e-ian) [inline authors/apps/articles/migrations/0001_initial.py]: You have unresolved merged conflicts here @anyatibrian

File: authors/apps/authentication/migrations/0001_initial.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
```
Comment (e-ian) [inline authors/apps/authentication/migrations/0001_initial.py]: Please resolve all conflicts before pushing for review

File: authors/apps/notification/views.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class NotificationView(generics.GenericAPIView):
+    permission_classes = [IsAuthenticated, ]
+    queryset = Notification.objects.all()
+    serializer_class = NotificationSerializer
+
+    def get(self, request):
+        user = Profile.objects.filter(user=request.user.id).first()
+        print(user.user.id)
```
Comment (e-ian) [inline authors/apps/notification/views.py]: @anyatibrian do we still need these print statements?

File: authors/apps/profiles/migrations/0001_initial.py [hunk 4]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
```
Comment (e-ian) [inline authors/apps/profiles/migrations/0001_initial.py]: Resolve conflicts

**Review by e-ian — COMMENTED** [review]
@anyatibrian I have left you a few comments on your `PR`

File: authors/apps/profiles/migrations/0001_initial.py (hunk 4 above)
Comment (anyatibrian) [reply authors/apps/profiles/migrations/0001_initial.py]: resolved

File: authors/apps/authentication/migrations/0001_initial.py (hunk 2 above)
Comment (anyatibrian) [reply authors/apps/authentication/migrations/0001_initial.py]: resolved

File: authors/apps/notification/models.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        fovarited = FavoriteArticle.objects.filter(favorited_by=instance.author.pk)
+        recipients = []
+        message = "{} comment on {}\n with \n {}".format(
+                    instance.author.user.username,
+                    instance.article.title,
+                    instance.body
+                )
+        for user in fovarited:
+            settings = NotificationSetting.objects.filter(user=user.favorited_by).first()
+            print(settings)
```
Comment (patrickf949) [inline authors/apps/notification/models.py]: Good work @anyatibrian 
Please remove the print statements though

**Review by patrickf949 — APPROVED** [review]
@anyatibrian LGTM

**Review by e-ian — APPROVED** [review]
Thanks for the great work Brian. LGTM

**Pull request merged**